src/general/WorldGenerationSettings.cs

```diff
@@ -1,11 +1,28 @@
 ﻿using System;
+using Godot;
+using Newtonsoft.Json;
 
 /// <summary>
 ///   Player configurable options for creating the game world
 /// </summary>
 public class WorldGenerationSettings
 {
-    private DifficultyPreset difficulty = null!;
+    [JsonConstructor]
+    public WorldGenerationSettings(IDifficulty difficulty)
+    {
+        if (difficulty is DifficultyPreset preset && preset.InternalName ==
+            SimulationParameters.Instance.GetDifficultyPreset("custom").InternalName)
+        {
+            GD.PrintErr(
+                $"Ignoring setting custom difficulty preset object to {nameof(WorldGenerationSettings)} " +
+                "(using normal instead). This should only happen when loading older saves");
```

Comment:
So loading an earlier save with custom settings would revert those settings to normal? I guess there's no easy way to avoid that?

Reply:
Yes. There's no easy way to avoid that. It would be possible to write a save upgrader which grabs the difficulty settings from the save and makes that into a new custom difficulty object. So while not very hard to do, it would take some extra effort to program and test.